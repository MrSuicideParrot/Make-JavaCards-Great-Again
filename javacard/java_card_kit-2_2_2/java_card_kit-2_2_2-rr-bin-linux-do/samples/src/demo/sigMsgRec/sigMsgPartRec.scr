
//+
// Copyright 2005 Sun Microsystems, Inc. All rights reserved.
// SUN PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
//-

//+
// Workfile:@(#)Header.scr	1.9
// Version:1.9
// Date:01/04/06
//-

powerup;

// Select the installer applet
0x00 0xA4 0x04 0x00 0x09 0xa0 0x00 0x00 0x00 0x62 0x03 0x01 0x08 0x01 0x7F;
// 90 00 = SW_NO_ERROR

// Turn output off during CAP file download
echo "CAP file download section. Output suppressed.";

output off;

0x80 0xB0 0x00 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/Header.cap
0x80 0xB2 0x01 0x00 0x00 0x7F;
0x80 0xB4 0x01 0x00 0x16 0x01 0x00 0x13 0xDE 0xCA 0xFF 0xED 0x01 0x02 0x04 0x00 0x01 0x09 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0C 0x7F;
0x80 0xBC 0x01 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/Directory.cap
0x80 0xB2 0x02 0x00 0x00 0x7F;
0x80 0xB4 0x02 0x00 0x20 0x02 0x00 0x1F 0x00 0x13 0x00 0x1F 0x00 0x0E 0x00 0x1F 0x00 0x7E 0x00 0x0C 0x02 0x24 0x00 0x96 0x00 0x63 0x00 0x00 0x01 0x05 0x00 0x06 0x00 0x03 0x00 0x83 0x03 0x7F;
0x80 0xB4 0x02 0x00 0x02 0x01 0x00 0x7F;
0x80 0xBC 0x02 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/Import.cap
0x80 0xB2 0x04 0x00 0x00 0x7F;
0x80 0xB4 0x04 0x00 0x20 0x04 0x00 0x1F 0x03 0x03 0x01 0x07 0xA0 0x00 0x00 0x00 0x62 0x01 0x01 0x03 0x01 0x07 0xA0 0x00 0x00 0x00 0x62 0x01 0x02 0x00 0x01 0x07 0xA0 0x00 0x00 0x00 0x62 0x7F;
0x80 0xB4 0x04 0x00 0x02 0x00 0x01 0x7F;
0x80 0xBC 0x04 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/Applet.cap
0x80 0xB2 0x03 0x00 0x00 0x7F;
0x80 0xB4 0x03 0x00 0x11 0x03 0x00 0x0E 0x01 0x0A 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0C 0x01 0x00 0x95 0x7F;
0x80 0xBC 0x03 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/Class.cap
0x80 0xB2 0x06 0x00 0x00 0x7F;
0x80 0xB4 0x06 0x00 0x0F 0x06 0x00 0x0C 0x00 0x80 0x03 0x06 0x00 0x04 0x07 0x01 0x00 0x00 0x00 0xA3 0x7F;
0x80 0xBC 0x06 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/Method.cap
0x80 0xB2 0x07 0x00 0x00 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x07 0x02 0x24 0x00 0x04 0x41 0x18 0x8C 0x00 0x1A 0x19 0x1E 0x25 0x29 0x04 0x16 0x04 0x61 0x08 0x18 0x8B 0x00 0x07 0x70 0x0C 0x18 0x19 0x1E 0x04 0x41 0x16 0x04 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x8B 0x00 0x0C 0x18 0x07 0x11 0x02 0x00 0x03 0x8D 0x00 0x0F 0x94 0x00 0x00 0x10 0x87 0x00 0x18 0x08 0x11 0x02 0x00 0x03 0x8D 0x00 0x0F 0x94 0x00 0x00 0x06 0x87 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x01 0xAD 0x01 0x7B 0x00 0x08 0x03 0x7B 0x00 0x08 0x92 0x8E 0x04 0x00 0x06 0x06 0xAD 0x01 0x7B 0x00 0x09 0x03 0x7B 0x00 0x09 0x92 0x8E 0x04 0x00 0x06 0x07 0xAD 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x00 0x7B 0x00 0x0A 0x03 0x7B 0x00 0x0A 0x92 0x8E 0x04 0x00 0x10 0x06 0xAD 0x00 0x7B 0x00 0x09 0x03 0x7B 0x00 0x09 0x92 0x8E 0x04 0x00 0x10 0x07 0x18 0x10 0x42 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x05 0x8D 0x00 0x0B 0x87 0x02 0x18 0x10 0x1E 0x03 0x8D 0x00 0x0D 0x94 0x00 0x00 0x0E 0x87 0x03 0x18 0x03 0x88 0x04 0x7A 0x05 0x30 0x8F 0x00 0x11 0x3D 0x18 0x1D 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x1E 0x8C 0x00 0x12 0x3B 0x7A 0x02 0x21 0x19 0x8B 0x00 0x13 0x2D 0x1A 0x03 0x25 0x61 0x0A 0x1A 0x04 0x25 0x10 0xA4 0x6B 0x03 0x7A 0x1A 0x04 0x25 0x73 0x00 0x22 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x00 0x10 0x00 0x12 0x00 0x0D 0x00 0x14 0x00 0x1B 0x18 0x19 0x8C 0x00 0x14 0x70 0x16 0x18 0x19 0x8C 0x00 0x15 0x70 0x0F 0x18 0x19 0x8C 0x00 0x16 0x70 0x08 0x11 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x6D 0x00 0x8D 0x00 0x17 0x7A 0x09 0x23 0x19 0x8B 0x00 0x13 0x2D 0x19 0x8B 0x00 0x18 0x32 0x04 0x05 0x8D 0x00 0x19 0x28 0x04 0xAD 0x03 0xAD 0x01 0x04 0x8E 0x03 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x00 0x0E 0x00 0x18 0xAD 0x03 0x1A 0x08 0x1F 0xAD 0x02 0x03 0x15 0x04 0x03 0x8E 0x08 0x00 0x0E 0x02 0x89 0x05 0xAD 0x02 0xAF 0x05 0x15 0x04 0x03 0x26 0x11 0xFF 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x00 0x53 0x10 0x08 0x4F 0x5B 0x38 0xAD 0x02 0xAF 0x05 0x04 0x41 0x15 0x04 0x03 0x26 0x11 0x00 0xFF 0x53 0x5B 0x38 0x19 0x8B 0x00 0x1B 0x3B 0x19 0xAF 0x05 0x05 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x41 0x8B 0x00 0x1C 0x19 0xAD 0x02 0x03 0xAF 0x05 0x05 0x41 0x8B 0x00 0x1D 0x7A 0x05 0x25 0xAD 0x03 0xAD 0x00 0x05 0x8E 0x03 0x00 0x0E 0x00 0x03 0x31 0x19 0x8B 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x00 0x13 0x2E 0x1B 0x07 0x25 0x11 0x00 0xFF 0x53 0x29 0x04 0x19 0x8B 0x00 0x18 0x29 0x05 0x1B 0x08 0xAD 0x02 0x03 0x16 0x05 0x8D 0x00 0x1E 0x3B 0xAD 0x03 0xAD 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x02 0x03 0x16 0x05 0x8E 0x04 0x00 0x0E 0x01 0x29 0x06 0xAD 0x03 0xAD 0x02 0x03 0x03 0x8E 0x04 0x00 0x0E 0x03 0x31 0x1E 0x61 0x08 0x11 0x91 0x02 0x8D 0x00 0x17 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x7A 0x05 0x24 0x19 0x8B 0x00 0x13 0x2D 0x1A 0x07 0x25 0x11 0x00 0xFF 0x53 0x32 0x19 0x8B 0x00 0x18 0x29 0x04 0x1A 0x08 0xAD 0x02 0x03 0x16 0x04 0x8D 0x00 0x1E 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x3B 0xAE 0x04 0x61 0x4B 0xAD 0x03 0xAD 0x00 0x05 0x8E 0x03 0x00 0x0E 0x00 0xAD 0x03 0xAD 0x02 0x03 0x16 0x04 0x8E 0x04 0x00 0x0E 0x01 0x29 0x05 0xAD 0x02 0x03 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x16 0x05 0x11 0xFF 0x00 0x53 0x10 0x08 0x4F 0x5B 0x38 0xAD 0x02 0x04 0x16 0x05 0x11 0x00 0xFF 0x53 0x5B 0x38 0x19 0x8B 0x00 0x1B 0x3B 0x19 0x05 0x8B 0x00 0x1C 0x7F;
0x80 0xB4 0x07 0x00 0x20 0x19 0xAD 0x02 0x03 0x05 0x8B 0x00 0x1D 0x18 0x04 0x88 0x04 0x70 0x1A 0x18 0x03 0x88 0x04 0xAD 0x03 0xAD 0x02 0x03 0x16 0x04 0x8E 0x04 0x00 0x0E 0x03 0x61 0x08 0x7F;
0x80 0xB4 0x07 0x00 0x07 0x11 0x91 0x02 0x8D 0x00 0x17 0x7A 0x7F;
0x80 0xBC 0x07 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/StaticField.cap
0x80 0xB2 0x08 0x00 0x00 0x7F;
0x80 0xB4 0x08 0x00 0x20 0x08 0x00 0x96 0x00 0x06 0x00 0x03 0x00 0x03 0x03 0x00 0x03 0x01 0x00 0x01 0x03 0x00 0x40 0xBE 0xDF 0xD3 0x7A 0x08 0xE2 0x9A 0x58 0x27 0x54 0x2A 0x49 0x18 0xCE 0x7F;
0x80 0xB4 0x08 0x00 0x20 0xE4 0x1A 0x60 0xDC 0x62 0x75 0xBD 0xB0 0x8D 0x15 0xA3 0x65 0xE6 0x7B 0xA9 0xDC 0x09 0x11 0x5F 0x9F 0xBF 0x29 0xE6 0xC2 0x82 0xC8 0x35 0x6B 0x0F 0x10 0x9B 0x19 0x7F;
0x80 0xB4 0x08 0x00 0x20 0x62 0xFD 0xBD 0x96 0x49 0x21 0xE4 0x22 0x08 0x08 0x80 0x6C 0xD1 0xDE 0xA6 0xD3 0xC3 0x8F 0x03 0x00 0x40 0x84 0x21 0xFE 0x0B 0xA4 0xCA 0xF9 0x7D 0xBC 0xFC 0x0E 0x7F;
0x80 0xB4 0x08 0x00 0x20 0xA9 0xBB 0x7A 0xBD 0x7D 0x65 0x40 0x2B 0x08 0xC6 0xDF 0xC9 0x4B 0x09 0x6A 0x29 0x3B 0xC2 0x42 0x88 0x23 0x44 0xAF 0x08 0x82 0x4C 0xFF 0x42 0xA4 0xB8 0xD2 0xDA 0x7F;
0x80 0xB4 0x08 0x00 0x19 0xCC 0xEE 0xC5 0x34 0xED 0x71 0x01 0xAB 0x3B 0x76 0xDE 0x6C 0xA2 0xCB 0x7C 0x38 0xB6 0x9A 0x4B 0x28 0x01 0x00 0x00 0x00 0x00 0x7F;
0x80 0xBC 0x08 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/ConstantPool.cap
0x80 0xB2 0x05 0x00 0x00 0x7F;
0x80 0xB4 0x05 0x00 0x20 0x05 0x00 0x7E 0x00 0x1F 0x02 0x00 0x00 0x00 0x02 0x00 0x00 0x01 0x02 0x00 0x00 0x03 0x02 0x00 0x00 0x02 0x02 0x00 0x00 0x05 0x02 0x00 0x00 0x04 0x01 0x81 0x08 0x7F;
0x80 0xB4 0x05 0x00 0x20 0x00 0x03 0x80 0x03 0x01 0x05 0x00 0x00 0x04 0x05 0x00 0x00 0x02 0x05 0x00 0x00 0x00 0x06 0x80 0x08 0x0D 0x03 0x80 0x03 0x02 0x06 0x81 0x0F 0x00 0x01 0x81 0x19 0x7F;
0x80 0xB4 0x05 0x00 0x20 0x00 0x06 0x81 0x0D 0x00 0x01 0x81 0x09 0x00 0x01 0x00 0x00 0x00 0x06 0x00 0x00 0x01 0x03 0x80 0x0A 0x01 0x06 0x00 0x00 0xE3 0x06 0x00 0x01 0x4D 0x06 0x00 0x01 0x7F;
0x80 0xB4 0x05 0x00 0x20 0x9E 0x06 0x80 0x07 0x01 0x03 0x80 0x0A 0x06 0x06 0x80 0x08 0x0F 0x06 0x80 0x03 0x00 0x03 0x80 0x0A 0x07 0x03 0x80 0x0A 0x09 0x03 0x80 0x0A 0x05 0x06 0x80 0x10 0x7F;
0x80 0xB4 0x05 0x00 0x01 0x02 0x7F;
0x80 0xBC 0x05 0x00 0x00 0x7F;

// com/sun/javacard/samples/sigMsgRec/javacard/RefLocation.cap
0x80 0xB2 0x09 0x00 0x00 0x7F;
0x80 0xB4 0x09 0x00 0x20 0x09 0x00 0x63 0x00 0x29 0x2E 0x0F 0x02 0x0F 0x0F 0x0F 0x16 0x0D 0x04 0x64 0x02 0x09 0x05 0x0B 0x02 0x02 0x0F 0x02 0x14 0x08 0x03 0x0A 0x02 0x20 0x09 0x02 0x0C 0x7F;
0x80 0xB4 0x09 0x00 0x20 0x02 0x2B 0x09 0x04 0x02 0x08 0x02 0x0C 0x0E 0x16 0x09 0x06 0x02 0x02 0x00 0x36 0x05 0x0D 0x0C 0x09 0x04 0x0B 0x04 0x07 0x04 0x05 0x06 0x04 0x05 0x06 0x04 0x05 0x7F;
0x80 0xB4 0x09 0x00 0x20 0x06 0x04 0x05 0x08 0x09 0x04 0x0C 0x07 0x08 0x23 0x07 0x07 0x08 0x07 0x05 0x06 0x0B 0x11 0x28 0x09 0x0B 0x0C 0x07 0x0E 0x0C 0x0C 0x0D 0x0B 0x07 0x0D 0x0C 0x0E 0x7F;
0x80 0xB4 0x09 0x00 0x06 0x0C 0x20 0x06 0x08 0x15 0x09 0x7F;
0x80 0xBC 0x09 0x00 0x00 0x7F;

0x80 0xBA 0x00 0x00 0x00 0x7F;
// resume printing output during functional test
output on;

//create applet With AID 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0C 0x01
echo "creating applet";
//0x80 0xB8 0x00 0x00 0x0D 0x0B 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x07 0x07 0x01 0x00 0x7F;
0x80 0xB8 0x00 0x00 0x0C 0x0A 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0C 0x01 0x00 0x7F;

// select applet         0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0C 0x01
0x00 0xA4 0x04 0x00 0x0A 0xA0 0x00 0x00 0x00 0x62 0x03 0x01 0x0C 0x0C 0x01 0x7F;
//expected 90 00

//send data 70 bytes
0x80 0x10 0x00 0x00 0x46 0x01 0x02 0x03 0x04 0x05 0x06 0x07 0x08 0x09 0x0a 0x0b 0x0c 0x0d 0x0e 0x0f 0x10 0x11 0x12 0x13 0x14 0x15 0x16 0x17 0x18 0x19 0x1a 0x1b 0x1c 0x1d 0x1e 0x1f 0x20 0x21 0x22 0x23 0x24 0x25 0x26 0x27 0x28 0x29 0x2a 0x2b 0x2c 0x2d 0x2e 0x2f 0x30 0x31 0x32 0x33 0x34 0x35 0x36 0x37 0x38 0x39 0x3a 0x3b 0x3c 0x3d 0x3e 0x3f 0x40 0x41 0x42 0x43 0x44 0x45 0x46 0x7F;
//Signature data expected. Last two bytes is the size of recoverable data. expected 42 bytes

//verify stage. send in signature
0x80 0x12 0x00 0x00 0x40 0x2d 0x15 0x79 0x89 0xba 0x71 0x6d 0x31 0x6c 0x0e 0x29 0x55 0xc0 0x0e 0x80 0xc3 0x5c 0xa3 0xe8 0xa1 0x12 0x65 0xe3 0x6f 0xb2 0x51 0x44 0x7d 0x30 0x4a 0x24 0xcf 0xa1 0x1b 0xaa 0x30 0x48 0xd3 0x70 0x4a 0x0b 0xe7 0x9a 0x05 0x1f 0x5f 0x87 0xc7 0x8f 0xe4 0xae 0xbc 0xde 0x0a 0x63 0x6a 0x28 0x48 0x52 0xc0 0xe7 0xd2 0x7f 0xfe 0x7f;
//expected output, the length of recoverable message. 0x2e

//complete signature verification by sending in non-recoverable part of message
0x80 0x12 0x00 0x00 0x1c 0x2b 0x2c 0x2d 0x2e 0x2f 0x30 0x31 0x32 0x33 0x34 0x35 0x36 0x37 0x38 0x39 0x3a 0x3b 0x3c 0x3d 0x3e 0x3f 0x40 0x41 0x42 0x43 0x44 0x45 0x46 0x7f;
//expected success 90 00



powerdown;

