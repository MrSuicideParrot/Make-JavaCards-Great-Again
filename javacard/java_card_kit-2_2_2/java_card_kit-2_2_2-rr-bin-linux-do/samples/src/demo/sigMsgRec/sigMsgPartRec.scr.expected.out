CLA: 00, INS: a4, P1: 04, P2: 00, Lc: 09, a0, 00, 00, 00, 62, 03, 01, 08, 01, Le: 00, SW1: 90, SW2: 00
CAP file download section. Output suppressed.
OUTPUT OFF;
OUTPUT ON;
creating applet
CLA: 80, INS: b8, P1: 00, P2: 00, Lc: 0c, 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0c, 01, 00, Le: 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0c, 01, SW1: 90, SW2: 00
CLA: 00, INS: a4, P1: 04, P2: 00, Lc: 0a, a0, 00, 00, 00, 62, 03, 01, 0c, 0c, 01, Le: 00, SW1: 90, SW2: 00
CLA: 80, INS: 10, P1: 00, P2: 00, Lc: 46, 01, 02, 03, 04, 05, 06, 07, 08, 09, 0a, 0b, 0c, 0d, 0e, 0f, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 1a, 1b, 1c, 1d, 1e, 1f, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 2a, 2b, 2c, 2d, 2e, 2f, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 3a, 3b, 3c, 3d, 3e, 3f, 40, 41, 42, 43, 44, 45, 46, Le: 42, 2d, 15, 79, 89, ba, 71, 6d, 31, 6c, 0e, 29, 55, c0, 0e, 80, c3, 5c, a3, e8, a1, 12, 65, e3, 6f, b2, 51, 44, 7d, 30, 4a, 24, cf, a1, 1b, aa, 30, 48, d3, 70, 4a, 0b, e7, 9a, 05, 1f, 5f, 87, c7, 8f, e4, ae, bc, de, 0a, 63, 6a, 28, 48, 52, c0, e7, d2, 7f, fe, 00, 2a, SW1: 90, SW2: 00
CLA: 80, INS: 12, P1: 00, P2: 00, Lc: 40, 2d, 15, 79, 89, ba, 71, 6d, 31, 6c, 0e, 29, 55, c0, 0e, 80, c3, 5c, a3, e8, a1, 12, 65, e3, 6f, b2, 51, 44, 7d, 30, 4a, 24, cf, a1, 1b, aa, 30, 48, d3, 70, 4a, 0b, e7, 9a, 05, 1f, 5f, 87, c7, 8f, e4, ae, bc, de, 0a, 63, 6a, 28, 48, 52, c0, e7, d2, 7f, fe, Le: 02, 00, 2a, SW1: 90, SW2: 00
CLA: 80, INS: 12, P1: 00, P2: 00, Lc: 1c, 2b, 2c, 2d, 2e, 2f, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 3a, 3b, 3c, 3d, 3e, 3f, 40, 41, 42, 43, 44, 45, 46, Le: 00, SW1: 90, SW2: 00
